### ISSUE TITLE
Disable `resolveEntityReferences` by default for newly created SAX parsers

### ISSUE THREAD

**Issue by philippn** [FasterXML/woodstox#211]
Hi there,

this is a follow-up issue for #209. With #209 the Woodstox SAX parser now generates `startEntity` and `endEntity` events just like Apache Xerces. In order to do so, one would need to manually call `getConfig().doReplaceEntityRefs(false)` on the STAX factory being passed to the `WstxSAXParserFactory` though.

This is especially troublesome as the normal procedure for obtaining a SAXParser would be by going through the JAXP discovery mechanism, which doesn't allow the configuration of this property.

I propose to modify the default constructor of the `WstxSAXParserFactory` so that it sets this property automatically. This would be enough for the JAXP discovery, which is probably the predominant use case. In contrast, I wouldn't touch the other constructor because that leaves the opportunity to change the behaviour to the old way, should this be necessary for some use cases.

I'm preparing a PR for this case and will submit it later today.

Thanks in advance and kind regards,
Philipp

**Comment by philippn** [comment]
A short follow-up: In #209 I had a mistake in my analysis regarding the different SAX parsers.

JDKs built-in Xerces produces this outcome:

> [main] INFO LoggingXMLFilter - Start element:  prodname prodname
> [main] INFO LoggingXMLFilter - Characters: eyePhone
> [main] INFO LoggingXMLFilter - Start entity: copyright
> [main] INFO LoggingXMLFilter - End entity: copyright
> [main] INFO LoggingXMLFilter - Characters: © 2.0
> [main] INFO LoggingXMLFilter - End element:  prodname prodname

This is a long standing issue with the JDKs Xerces fork, see: https://bugs.openjdk.org/browse/JDK-8331888

So the way it's handled in Woodstox now, is still correct I would say.

**Comment by cowtowncoder** [comment]
Ok, not sure I follow wrt whether #212 should go in as suggested... as in, does this:

    "the way it's handled in Woodstox now, is still correct I would say."

refer to state in 7.0.0, pre-PR, or for changed state with PR?

**Comment by philippn** [comment]
Hey @cowtowncoder,

the comment was referring to the post-change state in Woodstox.

Once the PR is merged, Woodstox will handle entities the same way as Apache Xerces does. The JDKs built-in Xerces fork differs slightly, but I would consider that a bug in the JDK (as it makes further processing almost impossible).

Hope that clears up the confusion.

Kind regards,
Philipp

**Comment by philippn** [comment]
To sum it up, this would be the outcome after the PR is merged:

JDK Xerces:

> [main] INFO LoggingXMLFilter - Start element: prodname prodname
> [main] INFO LoggingXMLFilter - Characters: eyePhone
> [main] INFO LoggingXMLFilter - Start entity: copyright
> [main] INFO LoggingXMLFilter - End entity: copyright
> [main] INFO LoggingXMLFilter - Characters: © 2.0
> [main] INFO LoggingXMLFilter - End element: prodname prodname

Apache Xerces and Woodstox:

> [main] INFO LoggingXMLFilter - Start element: meta meta
> [main] INFO LoggingXMLFilter - Start element: prodname prodname
> [main] INFO LoggingXMLFilter - Characters: eyePhone
> [main] INFO LoggingXMLFilter - Start entity: copyright
> [main] INFO LoggingXMLFilter - Characters: ©
> [main] INFO LoggingXMLFilter - End entity: copyright
> [main] INFO LoggingXMLFilter - Characters: 2.0
> [main] INFO LoggingXMLFilter - End element: prodname prodname

**Comment by cowtowncoder** [comment]
Thank you @philippn -- that does indeed clarify the situation, and I concur with your assessment.

**Comment by cowtowncoder** [comment]
Fixed via #212
